ui/src/app/applications/components/applications-list/applications-filter.tsx

```diff
@@ -1,238 +1,172 @@
-                                    />
-                                </li>
-                            </ul>
-                        </div>
-                    </div>
-                </div>
-            </div>
-        );
-    }
+const ProjectFilter = (props: AppFilterProps) => {
+    const [projects, loading, error] = useData(() => services.projects.list('items.metadata.name'), null, () => null);
+    const projectOptions = (projects || []).map(proj => {
+        return {label: proj.metadata.name};
+    });
+    return (
+        <Filter
+            label='PROJECTS'
+            selected={props.pref.projectsFilter}
+            setSelected={s => props.onChange({...props.pref, projectsFilter: s})}
+            field={true}
+            options={projectOptions}
+            error={error.state}
+            loading={loading}
+        />
+    );
+};
 
-    private getClusterDetail(dest: models.ApplicationDestination): string {
-        const cluster = this.props.clusters.find(target => target.name === dest.name || target.server === dest.server);
+const ClusterFilter = (props: AppFilterProps) => {
+    const getClusterDetail = (dest: ApplicationDestination, clusterList: Cluster[]): string => {
+        const cluster = (clusterList || []).find(target => target.name === dest.name || target.server === dest.server);
         if (!cluster) {
             return dest.server || dest.name;
         }
         if (cluster.name === cluster.server) {
             return cluster.name;
         }
         return `${cluster.name} (${cluster.server})`;
-    }
-}
+    };
+
+    const [clusters] = useData(() => services.clusters.list());
+    const clusterOptions = optionsFrom(
+        Array.from(new Set(props.apps.map(app => getClusterDetail(app.spec.destination, clusters)).filter(item => !!item))),
+        props.pref.clustersFilter
+    );
+
+    return (
+        <Filter label='CLUSTERS' selected={props.pref.clustersFilter} setSelected={s => props.onChange({...props.pref, clustersFilter: s})} field={true} options={clusterOptions} />
+    );
+};
+
+const NamespaceFilter = (props: AppFilterProps) => {
+    const namespaceOptions = optionsFrom(Array.from(new Set(props.apps.map(app => app.spec.destination.namespace).filter(item => !!item))), props.pref.namespacesFilter);
+    return (
+        <Filter
+            label='NAMESPACES'
+            selected={props.pref.namespacesFilter}
+            setSelected={s => props.onChange({...props.pref, namespacesFilter: s})}
+            field={true}
+            options={namespaceOptions}
+        />
+    );
+};
+
+export const ApplicationsFilter = (props: AppFilterProps) => {
+    const [hidden, setHidden] = React.useState(false);
+    return (
+        <React.Fragment>
+            <div className='applications-list__filters__title'>
+                FILTERS <i className='fa fa-filter' />
+                <ActionButton
+                    label={hidden ? 'SHOW' : 'HIDE'}
```

Comment:
Small bug:

1. Reduce page width and hide filter 
2. Expand page width

Actual behavior: Filter remains hidden 
Expected behavior: Filter should be visible

https://user-images.githubusercontent.com/426437/124004370-c6452080-d98c-11eb-8aad-6ce5dd9d393e.mov